pkg/provisioner/ironic/testserver/ironic.go

```diff
@@ -66,7 +78,12 @@ func (m *IronicMock) WithDrivers() *IronicMock {
 
 // WithNode configures the server with a valid response for /v1/nodes
 func (m *IronicMock) WithNode(node nodes.Node) *IronicMock {
-	m.ResponseJSON("/v1/nodes/"+node.UUID, node)
+	if node.UUID != "" {
+		m.ResponseJSON("/v1/nodes/"+node.UUID, node)
+	}
+	if node.Name != "" {
```

Comment:
I have a couple of question on this change: 
- Is it a valid case to have the UUID unset and Name set?
- When both UUID and Name are set, will the ServeMux.Handle panic?

Reply:
A real ironic service wouldn't have a node without a UUID, but in the tests we sometimes want the node to only be available via one or the other value to simulate the node having settings different from what the provisioner expects.

I'm not sure why the handler would panic? The endpoints are different (one looks up by UUID, the other by name).

Reply:
In case Name == UUID, but I think it's a trivial case

Reply:
Ah, yes, I didn't consider that. I think it's OK for the test to panic in that situation, since that doesn't seem like something we would expect to encounter in real data. I could add a test to protect against it, if you think it's worth it?

Reply:
I don't think it's worth it, and agree it's fine to panic in such situation